daemon/graphdriver/overlay2/overlay.go

```diff
@@ -312,23 +322,33 @@ func (d *Driver) Cleanup() error {
 // CreateReadWrite creates a layer that is writable for use as a container
 // file system.
 func (d *Driver) CreateReadWrite(id, parent string, opts *graphdriver.CreateOpts) error {
+	if opts != nil && len(opts.StorageOpt) != 0 && !projectQuotaSupported {
+		return fmt.Errorf("--storage-opt is supported only for overlay over xfs with 'pquota' mount option")
+	}
+
+	if opts == nil {
+		opts = &graphdriver.CreateOpts{}
+	}
+
+	if len(opts.StorageOpt) == 0 {
+		opts.StorageOpt = map[string]string{
+			"size": fmt.Sprintf("%d", d.options.quota.Size),
```

Comment:
Ah ok.
In Create, the default quota size is in d.options.quota.Size, where-as driver.quota.Size is initialized by the Create function to capture anything in "opts" param. In ReadOnly cases the opts will not have the "size" option since it does not go through the CreateReadWrite pathway.